app/controllers/api/solutions/original_solution_files_controller.rb

```diff
@@ -0,0 +1,13 @@
+module API
+  module Solutions
+    class OriginalSolutionFilesController < BaseController
+      def index
+        solution = current_user.solutions.find_by!(uuid: params[:solution_id])
```

Comment:
See top of api/files_controller for how we do this line elsewhere.

```suggestion
      begin
        solution = Solution.find_by!(uuid: params[:solution_id])
      rescue ActiveRecord::RecordNotFound
        return render_solution_not_found
      end
      
      return render_403(:solution_not_accessible) unless current_user.may_view_solution?(solution)
```